cluster-stamp.bicep

```diff
@@ -0,0 +1,1784 @@
+          'azure-arc'
+        ]
+      }
+      effect: {
+        value: 'audit'
+      }
+    }
+  }
+  dependsOn: []
+}
+
+resource paEnforceResourceLimits 'Microsoft.Authorization/policyAssignments@2021-06-01' = {
+  name: policyAssignmentNameEnforceResourceLimits
+  dependsOn: []
+  properties: {
+    displayName: '[${clusterName}] ${reference(policyResourceIdEnforceResourceLimits, '2020-09-01').displayName}'
+    scope: subscriptionResourceId('Microsoft.Resources/resourceGroups', resourceGroup().name)
+    policyDefinitionId: policyResourceIdEnforceResourceLimits
+    parameters: {
+      cpuLimit: {
+        value: '1000m'
+      }
+      memoryLimit: {
+        value: '512Mi'
+      }
+      excludedNamespaces: {
+        value: [
+          'kube-system'
+          'gatekeeper-system'
+          'azure-arc'
+          'cluster-baseline-settings'
+          'flux-system'
+        ]
+      }
+      effect: {
+        value: 'deny'
+      }
+    }
+  }
+}
+
+resource paEnforceImageSource 'Microsoft.Authorization/policyAssignments@2021-06-01' = {
+  name: policyAssignmentNameEnforceImageSource
+  properties: {
+    displayName: '[${clusterName}] ${reference(policyResourceIdEnforceImageSource, '2020-09-01').displayName}'
+    scope: subscriptionResourceId('Microsoft.Resources/resourceGroups', resourceGroup().name)
+    policyDefinitionId: policyResourceIdEnforceImageSource
+    parameters: {
+      allowedContainerImagesRegex: {
+        value: '${defaultAcrName}.azurecr.io/.+$|mcr.microsoft.com/.+$|azurearcfork8s.azurecr.io/azurearcflux/images/stable/.+$|docker.io/weaveworks/kured.+$|docker.io/library/.+$'
+      }
+      excludedNamespaces: {
+        value: [
+          'kube-system'
+          'gatekeeper-system'
+          'azure-arc'
+        ]
+      }
+      effect: {
+        value: 'deny'
+      }
+    }
+  }
+  dependsOn: []
+}
+
+resource paEnforceDefenderInCluster 'Microsoft.Authorization/policyAssignments@2021-06-01' = {
+  name: policyAssignmentNameEnforceDefenderInCluster
+  properties: {
+    displayName: '[${clusterName}] ${reference(policyResourceIdEnforceDefenderInCluster, '2020-09-01').displayName}'
+    description: 'Microsoft Defender for Containers should be enabled in the cluster.'
+    scope: subscriptionResourceId('Microsoft.Resources/resourceGroups', resourceGroup().name)
+    policyDefinitionId: policyResourceIdEnforceDefenderInCluster
+    parameters: {
+      effect: {
+        value: 'Audit'
+      }
+    }
+  }
+  dependsOn: []
+}
+
+resource miClusterControlPlane 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'mi-${clusterName}-controlplane'
+  location: location
+}
+
+resource miAppGatewayFrontend 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'mi-appgateway-frontend'
+  location: location
+}
+
+resource podmiIngressController 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'podmi-ingress-controller'
+  location: location
+}
```

Comment:
good catch @ckittel 

done | addressed from fcf99a9a62dd8a2f7f6bc4108ebea52ec8b63ef2

Reply:
Not sure I'm following what change you made here relative to the ask.  I see a bunch of `//` and only on some resources in this file.  Maybe that's all of them?  But it doesn't use `@description()`.  Ping me if there are any questions on this ask.

Reply:
I thought `@description()` was for params-only @ckittel we replace `//` by `@description()` when suitable.

done | addressed from 22e2afae6c0f57375a846c446ed08c53e34912ba

Reply:
Seems like we're still missing comments.  E.g. "Bootstraps your cluster using content from your repo" doesn't exist in the bicep file.  Are you sure you got them all moved over?

Reply:
Nevermind about that, I'm a bad searcher!

Reply:
I've just did another pass and all looks good @ckittel :muscle:

Reply:
we are unresolving this one since the following error @ckittel 

`Error BCP127: Function "description" cannot be used as a resource decorator.`

reverted | addressed from dad6fe32ad1b4b125ff86b01a429133f066cb720

Reply:
Oh, that must be new in the tooling.  BUMMER!  Thanks!